### PR TITLE
Add more server highlight label parameters

### PR DESCRIPTION
**Pull request by mhugent**
This PR adds new parameters to better control the placement of the highlight labels in the WMS server:
HIGHLIGHT_LABELDISTANCE : distance between feature (e.g. point or line) and the label in mm
HIGHLIGHT_LABELROTATION : label rotation in degrees
HIGHLIGHT_LABELVALI, HIGHLIGHT_LABELHALI : these parameters can be used to place the label directly on a point using the specified alignmant

### CODE REVIEW

File: src/server/services/wms/qgswmsparameters.cpp [hunk 1]
```diff
@@ -323,6 +323,18 @@ namespace QgsWms
     const QgsWmsParameter pHighlightBufferSize( QgsWmsParameter::HIGHLIGHT_LABELBUFFERSIZE );
     save( pHighlightBufferSize );
 
+    const QgsWmsParameter pLabelRotation( QgsWmsParameter::HIGHLIGHT_LABELROTATION, QVariant::Double );
+    save( pLabelRotation );
+
+    const QgsWmsParameter pLabelDistance( QgsWmsParameter::HIGHLIGHT_LABELDISTANCE, QVariant::Double );
+    save( pLabelDistance );
+
+    const QgsWmsParameter pLabelHali( QgsWmsParameter::HIGHLIGHT_LABELHALI );
```
Comment (elpaso) [inline src/server/services/wms/qgswmsparameters.cpp:332]: What do you think about using longer more readable identifiers? Something like `HIGHLIGHT_LABELHORIZONTALALIGNMENT`.

I would really prefer the even more readable form `HIGHLIGHT_LABEL_HORIZONTAL_ALIGNMENT` but I see that there are pre-existing compact forms `HIGHLIGHT_LABELBUFFERSIZE` without the`_`, maybe you could deprecate the pre-existing form and go for the better readability on all parameters.

File: src/server/services/wms/qgswmsparameters.cpp [hunk 2]
```diff
@@ -1634,6 +1666,10 @@ namespace QgsWms
     QStringList fonts = highlightLabelFont();
     QList<QColor> bufferColors = highlightLabelBufferColorAsColor();
     QList<double> bufferSizes = highlightLabelBufferSizeAsFloat();
+    QList<double> rotation = highlightLabelRotation();
```
Comment (elpaso) [inline src/server/services/wms/qgswmsparameters.cpp:1669]: Maybe const?

```suggestion
    const QList<double> rotation = highlightLabelRotation();
```

**Review by elpaso — COMMENTED** [review]
LGTM, just left some minor comments.

File: src/server/services/wms/qgswmsparameters.cpp (hunk 1 above)
Comment (mhugent) [reply src/server/services/wms/qgswmsparameters.cpp:332]: Yes, I was looking at the already existing parameter names. No strong opinion from my side if it should be HIGHLIGHT_LABELHORIZONTALALIGNMENT or HIGHLIGHT_LABEL_HORIZONTAL_ALIGNMENT.

Comment (elpaso) [reply src/server/services/wms/qgswmsparameters.cpp:332]: The latter is more readable, but I'm leaving the choice to you. 

If you think they are too long, a decent compromise could be something like `HIGHLIGHT_LABEL_HORIZ_ALIGN`.

Comment (mhugent) [reply src/server/services/wms/qgswmsparameters.cpp:332]: I'll go with HIGHLIGHT_LABEL_HORIZONTAL_ALIGNMENT

File: src/server/services/wms/qgswmsparameters.h [hunk 3]
```diff
@@ -1151,6 +1159,30 @@ namespace QgsWms
...
+       * Returns HIGHLIGHT_LABEL_DISTANCE as a list of double.
+       * \returns highlight label distance
+       */
+      QList<double> highlightLabelDistance() const;
+
+      /**
+       * Returns HIGHLIGHT_LABEL_HORIZONTAL_ALIGNMENT as a list of string.
+       * \returns highlight label hali
+       */
+      QStringList highlightLabelHali() const;
```
Comment (elpaso) [inline src/server/services/wms/qgswmsparameters.h:1178]: Please rename the functions.

File: src/server/services/wms/qgswmsparameters.h [hunk 4]
```diff
@@ -1151,6 +1159,30 @@ namespace QgsWms
...
+
+      /**
+       * Returns HIGHLIGHT_LABEL_VERTICAL_ALIGNMENT as a list of string.
+       * \returns highlight label vali
+       */
+      QStringList highlightLabelVali() const;
```
Comment (elpaso) [inline src/server/services/wms/qgswmsparameters.h:1184]: Rename.

**Review by elpaso — APPROVED** [review]
Looks good, except some minor documentation and method/var naming.

Feel free to merge when ready.

**Pull request merged**